app/src/full/java/io/homeassistant/companion/android/CrashHandling.kt

```diff
@@ -0,0 +1,33 @@
+package io.homeassistant.companion.android
+
+import android.content.Context
+import io.sentry.android.core.SentryAndroid
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+import javax.net.ssl.SSLException
+import javax.net.ssl.SSLHandshakeException
+import javax.net.ssl.SSLProtocolException
+
+fun initCrashReporting(context: Context) {
+    // Don't init on debug builds
+    if (BuildConfig.DEBUG)
+        return
+
+    SentryAndroid.init(context) { options ->
+        options.isEnableSessionTracking = true
+        options.isEnableNdk = false
+        options.dsn = "https://2d646f40f9574e0b9579e301a69bb030@o427061.ingest.sentry.io/5372876"
+        options.setBeforeSend { event, hint ->
+            return@setBeforeSend when (hint) {
+                is ConnectException,
+                is SocketTimeoutException,
+                is SSLException,
+                is SSLHandshakeException,
+                is SSLProtocolException,
+                is UnknownHostException -> null
+                else -> event
```

Comment:
Should we include `HTTPException`? Like: https://sentry.io/organizations/home-assistant-io/issues/1823402301/?project=5372876&query=is%3Aunresolved

Sentry also has their ANR issues that I am not sure if we need them? https://sentry.io/organizations/home-assistant-io/issues/1863659368/?project=5372876&query=is%3Aunresolved

Reply:
> Should we include `HTTPException`?

Because of how the dependencies are setup we can't right now, but we can explore in the future.

> Sentry also has their ANR issues that I am not sure if we need them?

ANR (App not responding) are events we should investigate further.  Most likely means we are doing something on the main thread we shouldn't be.